front/lib/api/assistant/actions/database_query.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  AgentConfigurationType,
+  AgentMessageType,
+  cloneBaseConfig,
+  DatabaseQueryActionType,
+  DatabaseQueryRunErrorEvent,
+  DatabaseQueryRunSuccessEvent,
+  DustProdActionRegistry,
+  isDatabaseQueryConfiguration,
+  ModelMessageType,
+  UserMessageType,
+} from "@dust-tt/types";
+
+import { runActionStreamed } from "@app/lib/actions/server";
+import { Authenticator } from "@app/lib/auth";
+import { AgentDatabaseQueryAction } from "@app/lib/models";
+import { generateModelSId } from "@app/lib/utils";
+
+export async function* runDatabaseQueryApp({
+  auth,
+  configuration,
+  userMessage,
+  agentMessage,
+}: {
+  auth: Authenticator;
+  configuration: AgentConfigurationType;
+  userMessage: UserMessageType;
+  agentMessage: AgentMessageType;
+}): AsyncGenerator<DatabaseQueryRunErrorEvent | DatabaseQueryRunSuccessEvent> {
+  // Checking authorizations
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Unexpected unauthenticated call to `runQueryDatabase`");
+  }
+  const c = configuration.action;
+  if (!isDatabaseQueryConfiguration(c)) {
+    throw new Error(
+      "Unexpected action configuration received in `runQueryDatabase`"
+    );
+  }
+  if (owner.sId !== c.dataSourceWorkspaceId) {
+    throw new Error("Unexpected datasource received in `runQueryDatabase`");
+  }
+
+  // Configuring and running the App
+  const config = cloneBaseConfig(
+    DustProdActionRegistry["assistant-v2-query-database"].config
+  );
+  const database = {
+    workspace_id: c.dataSourceWorkspaceId,
+    data_source_id: c.dataSourceId,
+    database_id: c.databaseId,
+  };
+  config.DATABASE_SCHEMA = {
+    type: "database_schema",
+    database,
+  };
+  config.DATABASE = {
+    type: "database",
+    database,
+  };
+
+  const res = await runActionStreamed(
+    auth,
+    "assistant-v2-query-database",
+    config,
+    [{ question: userMessage.content }]
```

Comment:
I think we want to rely on a paramter generation phase as is the case for DustAppRun

Otherwise it won't handle the case:

```
> What is the number of open projects
==> wront answer
> I meant the active ones
```